### DISCUSSION TITLE
Output a grid file in the format of a structured mesh

### DISCUSSION THREAD

**Discussion by ijku** [neperfepx/neper#735] [Q&A]
```
neper -T -n 1000 -id 0 -regularization 0 -domain 0.001,0.001,0.001 -o {neper_path}/domain -format tess,obj,ori
neper -T -loadtess {neper_path}/domain.tess -statcell x,y,z,vol,facelist -statface x,y,z,area
neper -M -rcl 1 -elttype hex -faset faces {neper_path}/domain.tess
```

Hi,
I employed a command similar to the one mentioned above to generate the grid file. Due to the domain being in the shape of a **cube** and being subdivided using **hexahedral** elements, the resulting grid is essentially a **structured mesh**. It doesn't require node-to-node connection information—**facemap**. Nevertheless, the resulting **.msh** file happens to be an unstructured mesh file that encloses **facemap**. 

Consequently, the resultant files are larger in size, which could lead to a slowdown in the program when producing time-varying grid output during subsequent computations.

In this case, Is there a way to output a structured grid format?

Thank you in advance,
Best,
Zixun

**Comment by rquey** [comment]
A mesh encloses the geometry (node coordinates) and connectivity (element definition from nodes).

I think that the `tesr` format would better suit your needs:
```
neper -T -n 1000 -domain "cube(0.001,0.001,0.001)" -o domain -format tesr
neper -V domain.tesr  -print img
```
![img](https://github.com/neperfepx/neper/assets/2659376/48d9c8dd-b41d-4ce4-bcb6-e724276767b2)

It is essentially a 3D grid.  Even more compact is the `raw` format.
